File: LeetCodeChallenges/DailyChallenges/Easy/TwoSum/python/soln.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_dict = {}

        for i, num in enumerate(nums):
            complement = target - num
            if complement in num_dict:
                return [num_dict[complement], i]
            num_dict[num] = i

two_sum = Solution().twoSum
print(two_sum([2, 7, 11, 15], 9))
print(two_sum([3, 2, 4], 6))
            
    

File: LeetCodeChallenges/DailyChallenges/Easy/TwoSum/JAVA/soln.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (numMap.containsKey(complement)) {
                return new int[] {numMap.get(complement), i};
            }
            numMap.put(nums[i], i);
        }

        throw new IllegalArgumentException("No two sum solution found.");
    }

    public static void main(String[] args) {
        Solution soln = new Solution();
        int[] nums = {2, 7, 11, 15};
        int target = 9;
        int[] result = soln.twoSum(nums, target);
        System.out.println(Arrays.toString(result));
    }
}